plugins/zenoh-plugin-storage-manager/src/replica/storage.rs

```diff
@@ -0,0 +1,248 @@
+
+    async fn start_storage_queryable_subscriber(&self) -> ZResult<Sender<StorageMessage>> {
+        self.initialize_if_empty().await;
+        let (tx, rx) = flume::bounded(1);
+
+        // subscribe on key_expr
+        let storage_sub = match self
+            .session
+            .declare_subscriber(&self.key_expr)
+            .res_async()
+            .await
+        {
+            Ok(storage_sub) => storage_sub,
+            Err(e) => {
+                error!("Error starting storage {} : {}", self.name, e);
+                return Err(e);
+            }
+        };
+
+        // answer to queries on key_expr
+        let storage_queryable = match self.session.declare_queryable(&self.key_expr).res().await {
+            Ok(storage_queryable) => storage_queryable,
+            Err(e) => {
+                error!("Error starting storage {} : {}", self.name, e);
+                return Err(e);
+            }
+        };
+
+        if self.replication.is_some() {
+            let aligner_updates = &self.replication.as_ref().unwrap().aligner_updates;
+            loop {
+                select!(
+                    // on sample for key_expr
+                    sample = storage_sub.recv_async() => {
+                        self.process_sample(sample.unwrap()).await;
+                    },
+                    // on query on key_expr
+                    query = storage_queryable.recv_async() => {
+                        self.reply_query(query).await;
+                    },
+                    // on aligner update
+                    update = aligner_updates.recv_async() => {
+                        match update {
+                            Ok(sample) => self.process_sample(sample).await,
+                            Err(e) => {
+                                error!("Error in receiving aligner update: {}", e);
+                                return Err(e.into());
+                            }
+                        }
+                    },
+                    // on storage handle drop
+                    message = rx.recv_async() => {
+                        match message {
+                            Ok(StorageMessage::Stop) => {
+                                trace!("Dropping storage {}", self.name);
+                                return Ok(tx);
+                            },
+                            Ok(StorageMessage::GetStatus(tx)) => {
+                                let storage = self.storage.lock().await;
+                                std::mem::drop(tx.send(storage.get_admin_status()).await);
+                                drop(storage);
+                            }
+                            Err(e) => {
+                                error!("Storage Message Channel Error: {}", e);
+                                return Err(e.into());
+                            },
+                        };
+                    }
+                );
+            }
+        } else {
+            loop {
```

Comment:
Same comment: this loop should be running in a side task.

Reply:
The previous solution is applicable here too.